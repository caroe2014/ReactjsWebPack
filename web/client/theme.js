// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
