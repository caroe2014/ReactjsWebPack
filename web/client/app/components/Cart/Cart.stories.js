// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs } from '@storybook/addon-knobs/react';

import { Provider } from 'rebass';

import theme from 'web/client/theme.js';
import Cart from './';
import config from 'app.config';
import { BrowserRouter as Router, Route } from 'react-router-dom';
import mockItems from './mockItems';

storiesOf('Cart', module)
  .addDecorator(withKnobs)
  .add('Cart Component', () => {
    return (
      <div style={{ textAlign: '-webkit-center' }}>
        <Provider theme={theme} style={{ width: '30em' }}>
          <Router basename={config.webPaths.base}>
            <Route render={({ match, location, history }) => (
              <Cart fillContent items={mockItems}/>
            )} />
          </Router>
        </Provider>
      </div>
    );
  });
