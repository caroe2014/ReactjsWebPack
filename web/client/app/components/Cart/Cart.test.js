// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { shallow, mount, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import Cart from '.';
import jsdom from 'jsdom';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';
import { t } from 'i18next';
import { createMemoryHistory } from 'history';

configure({
  adapter: new Adapter()
});

/* global describe, it, expect, beforeEach */
/* eslint-disable max-len */
global.window = new jsdom.JSDOM().window;
global.document = window.document;

configure({
  adapter: new Adapter()
});

jest.mock('i18next');
t.mockImplementation((key) => {
  return key;
});

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));
// jest.mock('react-i18next', () => ({
//   // this mock makes sure any components using the translate HoC receive the t function as a prop
//   withTranslation: () => Component => {
//     Component.defaultProps = { ...Component.defaultProps, t: (key) => key };
//     return Component;
//   },
// }));

const history = createMemoryHistory('/payments');

const props = {
  subTotal: 0,
  tax: 0,
  total: 0,
  items: [],
  fillContent: true
};

let mockFn = jest.fn();

let mockCurrency = {
  currencyDecimalDigits: '2',
  currencyCultureName: 'en-US',
  currencyCode: 'USD',
  currencySymbol: '$'
};

describe('Cart', () => {

  it('does a snapshot test', () => {
    const wrapper = shallow(<Cart {...props}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(shallow(<Cart {...props} />).exists()).toBe(true);
  });

  it('should render without throwing an error on mount', () => {
    expect(mount(<ThemeProvider theme={theme}><Cart {...props} changePending/></ThemeProvider>).exists()).toBe(true);
  });

  it('should render withnecessary props on mount ', () => {
    props.items = [
      {
        displayText: 'test',
        contextId: 100,
        selectedModifiers: [
          {
            description: 'test',
            baseAmount: 10,
            suboption: [
              {
                description: 'test',
                amount: 1
              }
            ]
          }
        ],
        count: 1,
        id: 1001,
        uniqueId: 1001,
        amount: 10
      }
    ];
    props.tipAmount = 10;
    props.total = 10;
    props.changePending = 0;
    props.storesList = [
      {
        id: 100,
        tip: {
          acceptTips: true
        }
      }
    ];
    props.currency = mockCurrency;
    props.fillContent = false;
    expect(mount(<ThemeProvider theme={theme}><Cart {...props} /></ThemeProvider>).exists()).toBe(true);
  });

  it('should render Loader when loading state is true', () => {
    let wrapper = shallow(<Cart {...props} />);
    wrapper.setState({ loader: true });
    wrapper.instance().forceUpdate();
    expect(wrapper.find('.loader-container').length).toBe(1);
  });

  it('should call handleDelete when DeleteButton is clicked', () => {
    let wrapper = shallow(<Cart {...props} removeItem={mockFn}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleDelete');
    wrapper.instance().forceUpdate();
    wrapper.find('.cart-delete-btn-1').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  describe('When price of modifier & sub modifier are 0', () => {
    it('should not show the price', () => {
      props.items = [
        {
          displayText: 'test',
          contextId: 100,
          selectedModifiers: [
            {
              description: 'test',
              baseAmount: 0,
              suboption: [
                {
                  description: 'test',
                  amount: 0
                }
              ]
            }
          ],
          count: 1,
          id: 1001,
          uniqueId: 1001,
          amount: 10
        }
      ];
      props.storesList = [
        {
          id: 100,
          tip: {
            acceptTips: true
          },
          pay: {
            paymentsEnabled: true
          }
        }
      ];
      props.changePending = 10;
      let wrapper = shallow(<Cart {...props} />);
      expect(wrapper.find('.modifier-empty-price').length).toBe(1);
      expect(wrapper.find('.sub-modifier-empty-price').length).toBe(1);
    });
  });

  it('should render withnecessary props on mount ', () => {
    let mockprops = {};
    mockprops.items = [
      {
        displayText: 'test',
        contextId: 100,
        count: 1,
        id: 1001,
        uniqueId: 1001,
        amount: 10
      }
    ];
    mockprops.tipAmount = 10;
    mockprops.total = 10;
    mockprops.changePending = 0;
    mockprops.storesList = [
      {
        id: 100,
        tip: {
          acceptTips: true
        },
        pay: {
          paymentsEnabled: true
        }
      }
    ];
    mockprops.currency = mockCurrency;
    mockprops.fillContent = false;
    expect(mount(<ThemeProvider theme={theme}><Cart {...mockprops} /></ThemeProvider>).exists()).toBe(true);
  });

  // it('should call handleCheckout when payments is disabled ', () => {
  //   let mockprops = {};
  //   mockprops.items = [
  //     {
  //       displayText: 'test',
  //       contextId: 100,
  //       count: 1,
  //       id: 1001,
  //       uniqueId: 1001,
  //       amount: 10
  //     }
  //   ];
  //   mockprops.tipAmount = 10;
  //   mockprops.total = 10;
  //   mockprops.changePending = 0;
  //   mockprops.storesList = [
  //     {
  //       id: 100,
  //       tip: {
  //         acceptTips: true
  //       },
  //       pay: {
  //         paymentsEnabled: false
  //       }
  //     }
  //   ];
  //   mockprops.currency = mockCurrency;
  //   mockprops.fillContent = false;
  //   let wrapper = shallow(<Cart {...mockprops} setCheckoutData={mockFn}/>);
  //   const spy = jest.spyOn(wrapper.instance(), 'handleCheckout');
  //   wrapper.instance().forceUpdate();
  //   wrapper.find('.pay-cart-button').simulate('click');
  //   expect(spy).toHaveBeenCalled();
  //   spy.mockClear();
  // });

  describe('handlePay clicked', () => {
    let mockprops = {};
    mockprops.items = [
      {
        displayText: 'test',
        contextId: 100,
        count: 1,
        id: 1001,
        uniqueId: 1001,
        amount: 10
      }
    ];
    mockprops.tipAmount = 10;
    mockprops.total = 10;
    mockprops.changePending = 0;
    mockprops.location = {
      pathname: '/test'
    };
    mockprops.storesList = [
      {
        id: 100,
        tip: {
          acceptTips: true
        },
        pay: {
          paymentsEnabled: true
        }
      }
    ];
    mockprops.currency = mockCurrency;
    mockprops.fillContent = false;

    // it('should call handlePay when payments is enabled ', () => {
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('return when location is payment ', () => {
    //   mockprops.location = {
    //     pathname: '/payment'
    //   };
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('accept tips is false ', () => {
    //   mockprops.location = {
    //     pathname: '/test'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('deliveryDestination is enabled ', () => {
    //   mockprops.location = {
    //     pathname: '/test'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('deliveryDestination is enabled but path is deliveryLocation ', () => {
    //   mockprops.location = {
    //     pathname: '/deliveryLocation'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('nameCapture is enabled ', () => {
    //   mockprops.location = {
    //     pathname: '/test'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: false
    //       },
    //       nameCapture: {
    //         featureEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('nameCapture is enabled but path is nameCapture ', () => {
    //   mockprops.location = {
    //     pathname: '/nameCapture'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: false
    //       },
    //       nameCapture: {
    //         featureEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('sms is enabled ', () => {
    //   mockprops.location = {
    //     pathname: '/test'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: false
    //       },
    //       nameCapture: {
    //         featureEnabled: false
    //       },
    //       sms: {
    //         isSmsEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('Sms is enabled but path is isSmsEnabled ', () => {
    //   mockprops.location = {
    //     pathname: '/smsNotification'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: false
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: false
    //       },
    //       nameCapture: {
    //         featureEnabled: false
    //       },
    //       sms: {
    //         isSmsEnabled: true
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });

    // it('Accept tips is true but path is tip ', () => {
    //   mockprops.location = {
    //     pathname: '/tip'
    //   };
    //   mockprops.storesList = [
    //     {
    //       id: 100,
    //       tip: {
    //         acceptTips: true
    //       },
    //       pay: {
    //         paymentsEnabled: true
    //       },
    //       deliveryDestination: {
    //         deliverToDestinationEnabled: false
    //       },
    //       nameCapture: {
    //         featureEnabled: false
    //       },
    //       sms: {
    //         isSmsEnabled: false
    //       }
    //     }
    //   ];
    //   let wrapper = shallow(<Cart {...mockprops} lastCartLocation={mockFn} setCheckoutData={mockFn} setOrderConfig={mockFn} setCartLoyaltyInfo={mockFn} history={history} toggleCart={mockFn} closeCart={mockFn}/>);
    //   const spy = jest.spyOn(wrapper.instance(), 'handlePay');
    //   wrapper.instance().forceUpdate();
    //   wrapper.find('.pay-cart-button').simulate('click');
    //   expect(spy).toHaveBeenCalled();
    //   spy.mockClear();
    // });
  });

});
