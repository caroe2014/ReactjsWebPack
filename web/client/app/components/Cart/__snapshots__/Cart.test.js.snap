// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart does a snapshot test 1`] = `
<Styled(Flex)
  alignItems="center"
  className="cart-container"
  flexDirection="column"
  m={3}
  width={1}
>
  <Styled(Flex)
    aria-label="CART_MY_CART"
    className="my-cart-header"
    role="header"
    tabIndex={0}
  >
    <div
      i18nKey="CART_MY_CART"
    />
  </Styled(Flex)>
  <Styled(Divider)
    className="horiz-border"
  />
  <Styled(Flex)>
    <Styled(Box)
      aria-label="CART_EMPTY"
      aria-live="polite"
      style={
        Object {
          "left": "0px",
          "position": "absolute",
          "right": "0px",
          "textAlign": "center",
          "top": "50%",
        }
      }
      tabIndex={1}
    >
      <div
        i18nKey="CART_EMPTY"
      />
    </Styled(Box)>
  </Styled(Flex)>
</Styled(Flex)>
`;
